src/main/app/src/app/core/websocket/websocket.service.ts

```diff
@@ -212,6 +220,51 @@ export class WebsocketService implements OnDestroy {
     });
   }
 
+  public longRunningOperation(
+    opcode: Opcode,
+    objectType: ObjectType,
+    objectId: string,
+    operation: () => Observable<void>,
+  ) {
+    return this.listenOnce(opcode, objectType, objectId).pipe(
+      switchMap((s) =>
+        s === subscriptionState.subscribed ? operation() : of(s),
+      ),
+      filter((x) => x === subscriptionState.received),
+    ) as Observable<void>;
```

Comment:
is this typecasting necessary? it does not look like this observable will return void